test/extensions/filters/network/wasm/wasm_filter_test.cc

```diff
@@ -190,6 +193,85 @@ TEST_P(WasmNetworkFilterTest, SegvFailOpen) {
   EXPECT_EQ(Network::FilterStatus::Continue, filter().onData(fake_downstream_data, false));
 }
 
+TEST_P(WasmNetworkFilterTest, RestrictOnNewConnection) {
+  if (std::get<0>(GetParam()) != "v8" || std::get<1>(GetParam()) != "cpp") {
+    return;
+  }
+  AllowedCapabilitiesMap allowed_capabilities = {
+      {"proxy_on_context_create", proxy_wasm::SanitizationConfig()},
+      {"proxy_get_property", proxy_wasm::SanitizationConfig()},
+      {"proxy_log", proxy_wasm::SanitizationConfig()},
+      {"proxy_on_new_connection", proxy_wasm::SanitizationConfig()}};
+  setupConfig("", "logging", false, allowed_capabilities);
+  setupFilter();
+
+  // Expect this call, because proxy_on_new_connection is allowed
+  EXPECT_CALL(filter(), log_(spdlog::level::trace, Eq(absl::string_view("onNewConnection 2"))));
+  EXPECT_EQ(Network::FilterStatus::Continue, filter().onNewConnection());
+
+  // Do not expect this call, because proxy_on_downstream_connection_close is not allowed
+  EXPECT_CALL(filter(),
+              log_(spdlog::level::trace, Eq(absl::string_view("onDownstreamConnectionClose 2 1"))))
+      .Times(0);
+  read_filter_callbacks_.connection_.close(Network::ConnectionCloseType::FlushWrite);
+  // Noop.
+  read_filter_callbacks_.connection_.close(Network::ConnectionCloseType::FlushWrite);
+  filter().testClose();
+}
+
+TEST_P(WasmNetworkFilterTest, RestrictOnDownstreamConnectionClose) {
+  if (std::get<0>(GetParam()) != "v8" || std::get<1>(GetParam()) != "cpp") {
```

Comment:
I think this test should work with not only v8 && cpp but also other runtimes and languages except  'null' plugin, though I haven't testes.

Reply:
Sorry for the delay. Fixed.